genesyscloud/resource_genesyscloud_integration.go

```diff
@@ -0,0 +1,495 @@
+package genesyscloud
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/platformclientv2"
+)
+
+var (
+	integrationConfigResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Integration config name.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"notes": {
+				Description: "Integration config notes.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"properties": {
+				Description: "Integration config properties.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"advanced": {
+				Description: "Integration advanced config.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"credentials": {
+				Description: "Credentials required for the integration. The required keys are indicated in the credentials property of the Integration Type.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+		},
+	}
+)
+
+func getAllIntegrations(ctx context.Context, clientConfig *platformclientv2.Configuration) (ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(ResourceIDMetaMap)
+	integrationAPI := platformclientv2.NewIntegrationsApiWithConfig(clientConfig)
+
+	for pageNum := 1; ; pageNum++ {
+		integrations, _, err := integrationAPI.GetIntegrations(100, pageNum, "", nil, "", "")
+		if err != nil {
+			return nil, diag.Errorf("Failed to get page of integrations: %v", err)
+		}
+
+		if integrations.Entities == nil || len(*integrations.Entities) == 0 {
+			break
+		}
+
+		for _, integration := range *integrations.Entities {
+			resources[*integration.Id] = &ResourceMeta{Name: *integration.Name}
+		}
+	}
+
+	return resources, nil
+}
+
+func integrationExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllIntegrations),
+		RefAttrs: map[string]*RefAttrSettings{
+			// TODO: Since now it uses jsonencode, might need to change how export works
+			"config.credentials.*": {RefType: "genesyscloud_integration_credentials"},
```

Comment:
config.credentials is just a string, so I don't think this is actually going to work as-is if it is JSON. You can probably just remove this reference for now.